src/helpers/fetchUrlsData.ts

```diff
@@ -1,11 +1,23 @@
+function fetchSingleUrlData(url: string): Promise<string> {
+  return fetch(url).then((r) => r.json());
+}
+
 export async function fetchUrlsData(
   urls: Record<string, string>,
 ): Promise<Record<string, string>> {
-  const data: Record<string, string> = {};
-  for (const [k, u] of Object.entries(urls)) {
-    await fetch(u).then(async (r) => {
-      data[k] = await r.json();
-    });
-  }
-  return data;
+  /* Convert URLs into an array of promises of tuples consisting of
+   * on one hand, the key of the URL,
+   * and on the other, the response to the URL itself
+   */
+  const keyAndResponsePromises = Object.entries(urls).map(
+    async ([key, url]): Promise<[string, string]> => [
+      key,
+      await fetchSingleUrlData(url),
+    ],
+  );
+  const keyAndResponses = await Promise.all(keyAndResponsePromises);
```

Comment:
as discussed in #92 this could probably be `allSettled`

Reply:
I'd rework this on its own PR because to do it right I'd have to modify the function signature and change the places where it's used

Reply:
we're under no release pressure (yet) let's do this right-the-first-time (c)

Reply:
Alright, but for the sake of respecting my own goals for this PR I'll do it on another branch

Reply:
I believe that we should merge this until App.js has been migrated to Typescript. See #96  for why.